core/types/transaction.go

```diff
@@ -188,11 +188,7 @@ func (tx *Transaction) CheckNonce() bool   { return true }
 // To returns the recipient address of the transaction.
 // It returns nil if the transaction is a contract creation.
 func (tx *Transaction) To() *common.Address {
-	if tx.data.Recipient == nil {
-		return nil
-	}
-	to := *tx.data.Recipient
-	return &to
+	return tx.data.Recipient
```

Comment:
the idea was to return a copy of recipient value and not the reference to the same recipient. why is it unnecessary?

Reply:
I see, it was not stated in the comment, which is  *To returns the recipient address of the transaction*, nor did I see it in the tests. why do we return a copy ? transaction's fields are already private.

Reply:
If you look at all the other fields, everything returns a copy. The idea is that we want to prevent requesting some data about a transaction and ten modifying the transaction from the outside.